Support for filters in the Druid Delta Lake connector (#16288)
apache/druid: docs/ingestion/input-sources.md

You can use the Delta input source to read data stored in a Delta Lake table. For a given table, the input source scans
the latest snapshot from the configured table. Druid ingests the underlying delta files from the table.

 | Property|Description|Required|
|---------|-----------|--------|
| type|Set this value to `delta`.|yes|
| tablePath|The location of the Delta table.|yes|
| filter|The JSON Object that filters data files within a snapshot.|no|

### Delta filter object

You can use these filters to filter out data files from a snapshot, reducing the number of files Druid has to ingest from
a Delta table. This input source provides the following filters: `and`, `or`, `not`, `=`, `>`, `>=`, `<`, `<=`.

When a filter is applied on non-partitioned columns, the filtering is best-effort as the Delta Kernel solely relies
on statistics collected when the non-partitioned table is created. In this scenario, this Druid connector may ingest

> data that doesn't match the filter. For guaranteed filtering behavior, only use filters on partitioned columns.

Reviewer: ```suggestion
data that doesn't match the filter. To guarantee that the Delta Kernel prunes out the unnecessary columns, only use filters on partitioned columns.
```
I think the wording can be slightly improved - "only" makes it seem like something unintuitive will happen when the filtering is done on a non-partitioned column. Unless it will degrade the performance, than if the filter was not present, we can probably reword it like a guideline.

Author: Nice, I like it 👍

| column   | The table column to apply the filter on. | yes      |
| value    | The value to use in the filter.          | yes      |

`>` filter:

| Property | Description                              | Required |
|----------|------------------------------------------|----------|
| type     | Set this value to `>`.                   | yes      |
| column   | The table column to apply the filter on. | yes      |
| value    | The value to use in the filter.          | yes      |

`>=` filter:

| Property | Description                              | Required |
|----------|------------------------------------------|----------|
| type     | Set this value to `>=`.                  | yes      |
| column   | The table column to apply the filter on. | yes      |
| value    | The value to use in the filter.          | yes      |

`<` filter:

| Property | Description                              | Required |
|----------|------------------------------------------|----------|
| type     | Set this value to `<`.                   | Yes      |
| column   | The table column to apply the filter on. | Yes      |
| value    | The value to use in the filter.          | Yes      |

`<=` filter:

> | Property | Description                              | Required |
> |----------|------------------------------------------|----------|
> | type     | Set this value to `<=`.                  | yes      |
> | column   | The table column to apply the filter on. | yes      |
> | value    | The value to use in the filter.          | yes      |

Reviewer: There is some redundancy in the docs. Perhaps, we can merge all these filters into a single table. 
```suggestion
| Property | Description                              | Required |
|----------|------------------------------------------|----------|
| type     | Set this value to '=', '>=', ....`<=`.                  | yes      |
| column   | The table column to apply the filter on. | yes      |
| value    | The value to use in the filter.          | yes      |
```

The following is a sample spec to read all records from the Delta table `/delta-table/foo`:

```json
...
    "ioConfig": {
      "type": "index_parallel",
      "inputSource": {
        "type": "delta",
        "tablePath": "/delta-table/foo"
      },
    }
```

> The following is a sample spec to read records from the Delta table `/delta-table/foo` that match the `and` filter
> to select records where `name = 'Employee4'` and `age >= 30`:

Reviewer: ```suggestion
The following is a sample spec to read records from the Delta table `/delta-table/foo` to select records where `name = 'Employee4'` 'and' `age >= 30`:
```
